server/mscalendar/availability_test.go

```diff
@@ -92,25 +92,14 @@ func TestSyncStatusAll(t *testing.T) {
 				},
 			}, nil).AnyTimes()
 
-			token := &oauth2.Token{
-				AccessToken: "bot_oauth_token",
-			}
-			s.EXPECT().LoadUser("bot_mm_id").Return(&store.User{
-				MattermostUserID: "bot_mm_id",
-				OAuth2Token:      token,
-				Remote: &remote.User{
-					ID:   "bot_remote_id",
-					Mail: "bot_email@example.com",
-				},
-			}, nil).Times(1)
-
-			mockPluginAPI.EXPECT().GetMattermostUser("bot_mm_id").Return(&model.User{}, nil).Times(1)
-
-			mockRemote.EXPECT().MakeClient(context.Background(), token).Return(mockClient).Times(1)
-			mockClient.EXPECT().GetSuperuserToken().Return("bot_bearer_token", nil)
-			mockRemote.EXPECT().MakeSuperuserClient(context.Background(), "bot_bearer_token").Return(mockClient)
+			mockRemote.EXPECT().MakeSuperuserClient(context.Background()).Return(mockClient, nil)
 
-			mockClient.EXPECT().GetSchedule("bot_remote_id", []string{"user_email@example.com"}, gomock.Any(), gomock.Any(), 15).Return([]*remote.ScheduleInformation{tc.sched}, nil)
+			mockClient.EXPECT().GetSchedule(gomock.Any(), gomock.Any(), gomock.Any(), 15).DoAndReturn(
```

Comment:
I am not a fan to use Any, but I understand this make the test simpler and not specially a worse test.

Reply:
These arguments are structs, so they cannot be asserted inline. Instead, we call `DoAndReturn`, which allows us to examine the args that were passed in, and do our own assertions within the callback. I've added some more assertions here now.